pkg/polylog/interface.go

```diff
@@ -0,0 +1,129 @@
+package polylog
+
+import "time"
+
+// TODO_CONSIDERATION: this may be a good candidate package for extraction to
+// its own module.
+
+// TODO_INVESTIGATE: check whether the pkg dependency tree includes all logging
+// libraries.
+
+// Logger is an interface that exposes methods for each supported log level, each
+// of which returns an Event.
+type Logger interface {
```

Comment:
You mentioned aiming for equivalence with cosmos/comet.

I know our level messaging is different to conform with zerlog, but thought we could consider adding `With(keyvals ...interface{}) Logger`

```go
// Logger is what any CometBFT library should take.
type Logger interface {
	Debug(msg string, keyvals ...interface{})
	Info(msg string, keyvals ...interface{})
	Error(msg string, keyvals ...interface{})

	With(keyvals ...interface{}) Logger
}
```

Reply:
:100: thanks for the suggestion! :mechanical_arm: I had thought about this but did not manage to write it down anywhere before getting distracted. :sweat_smile:

Reply:
There may be additional parts of the zerolog API that we which to emulate/expose. Keep an :eye:  out.